File: 백준/Silver/21921. 블로그/블로그.java
import java.util.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int X = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());

        int[] nums = new int[N];
        for(int i = 0; i < N; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }

        int right = 0;
        int left = 0;
        int visitMax = 0;
        int count = 0;
        int visit = nums[0];
        
        while(right < N && left < N) {
            if(right - left + 1 == X) {
                if(visitMax == visit) {
                    count++;
                } 
                else if(visitMax < visit) {
                    visitMax = visit;
                    count = 1;
                }

                visit -= nums[left];
                left++;
            }
            else {
                right++;

                if(right < N) {
                    visit += nums[right];
                }
            }
        }

        if(visitMax == 0) {
            System.out.println("SAD");
        }
        else {
            System.out.println(visitMax);
            System.out.println(count);
        }
    }
}

File: 백준/Silver/21921. 블로그/블로그.py
import sys
input = sys.stdin.readline

N, X = map(int, input().split())

nums = list(map(int, input().split()))

max_visit = 0
count = 0
visit = nums[0]

right = 0
left = 0

while right < N and left < N:
    if right - left + 1 == X:
        if max_visit == visit:
            count += 1
        elif max_visit < visit:
            max_visit = visit
            count = 1

        visit -= nums[left]
        left += 1

    else:
        right += 1
        if right < N:
            visit += nums[right]

if max_visit == 0:
    print('SAD')
else:
    print(max_visit)
    print(count)